http-serverless/src/main/java/io/micronaut/http/serverless/RawHttpBasedServletHttpRequest.java

```
@@ -0,0 +1,216 @@
+package io.micronaut.http.serverless;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.convert.value.MutableConvertibleValues;
+import io.micronaut.core.convert.value.MutableConvertibleValuesMap;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.HttpParameters;
+import io.micronaut.http.cookie.Cookies;
+import io.micronaut.servlet.http.ServletHttpRequest;
+import rawhttp.cookies.ServerCookieHelper;
+import rawhttp.core.RawHttp;
+import rawhttp.core.RawHttpHeaders;
+import rawhttp.core.RawHttpRequest;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Sahoo.
+ */
+class RawHttpBasedServletHttpRequest implements ServletHttpRequest<RawHttpRequest, Object> {
+    private final RawHttp rawHttp;
+    private final InputStream in;
+    private final RawHttpRequest rawHttpRequest;
+    private final RawHttpBasedHeaders headers;
+
+    private final RawHttpBasedParameters queryParameters;
+
+    public RawHttpBasedServletHttpRequest(InputStream in, ConversionService conversionService) {
+        this.rawHttp = new RawHttp();
+        this.in = in;
+        try {
+            rawHttpRequest = rawHttp.parseRequest(in);
+//            System.err.println("DEBUG: Parsed following request from input stream: \n" + rawHttpRequest);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

need to throw a better exception